frontend/src/components/modals/AddRoadmapModal.tsx

```diff
@@ -17,18 +23,87 @@ import {
   DisplayInvitedMember,
   AddOrModifyMember,
   SkipMemberAddition,
+  isInviteUser,
 } from './modalparts/TeamMemberModalParts';
+import { AddOrModifyCustomer } from './modalparts/CustomerModalParts';
+import { Permission } from '../../../../shared/types/customTypes';
 import css from './AddRoadmapModal.module.scss';
 import { apiV2 } from '../../api/api';
+import { hasPermission } from '../../../../shared/utils/permission';
 
 const classes = classNames.bind(css);
 
+const removeRepresentative = (
+  customers: RoadmapCreationCustomer[],
+  member: InviteRoadmapUser,
+) =>
+  customers.map((customer) => ({
+    ...customer,
+    representatives: customer.representatives.filter(
+      ({ email }) => email !== member.email,
+    ),
+  }));
+
+const getKey = (person: InviteRoadmapUser | RoadmapCreationCustomer) =>
+  isInviteUser(person) ? person.email : `${person.name}-${person.email}`;
```

Comment:
```suggestion
  isInviteUser(person) ? person.email : `${person.name}@${person.email}`;
```

`-` is ambiguous separator, as it can appear any number of times on either side, e.g.
name = "a-b", email = "c@d"
name = "a", email = "b-c@d"

`@` would work, as it is guaranteed to appear exactly once in a valid email, also any character that is not allowed in email address would work